### PR TITLE
Update usaco-1114.mdx

### PR DESCRIPTION
**Pull request by TruaShamu**
1. Replaced original cpp code with  new code by @nwatx (old one WAs on all testcases except for the first) 
2. Added java code

### CODE REVIEW

File: solutions/usaco-1114.mdx [hunk 1]
```diff
@@ -37,37 +37,88 @@ $$
 ```cpp
 //CodeSnip{CPP Short Template}
```
Comment (nwatx) [inline solutions/usaco-1114.mdx:38]: add your name to authors

**Comment by TruaShamu** [comment]
@nwatx  The current cpp code on the internal sol gives WA, so I replaced it with the one you dmed me on discord, if that's okay with you?

File: solutions/usaco-1114.mdx (hunk 1 above)
Comment (TruaShamu) [reply solutions/usaco-1114.mdx:38]: ok, done

**Comment by nwatx** [comment]
> @nwatx The current cpp code on the internal sol gives WA, so I replaced it with the one you dmed me on discord, if that's okay with you?

yes, and I did make a commit fixing this earlier (so I think if you test out the code currently on the website it should still pass)

**Review by nwatx — APPROVED** [review]

File: solutions/usaco-1114.mdx [hunk 2]
```diff
@@ -37,37 +37,88 @@ $$
...
+<JavaSection>
+	
+```java
+import java.io.*;
+import java.util.*;
+
+public class ModernArt3 {
+public static final int MAXN = 301;
+public static int[] inp = new int[MAXN];
+public static int[][] DP = new int[MAXN][MAXN];
```
Comment (SansPapyrus683) [inline solutions/usaco-1114.mdx:85]: 1. indentation is really weird
2. rename `inp` to like `painting` or smth
3. rename `DP` to `dp`

File: solutions/usaco-1114.mdx [hunk 3]
```diff
@@ -37,37 +37,88 @@ $$
...
+			for (int j = 0; j <= N; j++) {
+				DP[i][j] = MAXN + 1;
+			}
+		}
+
+		for(int i = 0; i < MAXN; i++) {
+			DP[i][i] = 1;
+		}
+
+		for(int s = 1; s <= N; s++) {
```
Comment (SansPapyrus683) [inline solutions/usaco-1114.mdx:105]: can you have spaces for the control statements
like
```suggestion
		for (int s = 1; s <= N; s++) {
```

File: solutions/usaco-1114.mdx [hunk 4]
```diff
@@ -37,37 +37,88 @@ $$
...
+			for(int i = 0; i <= N - s; i++) {
+				for(int j = i; j < i + s; j++) {
+					int k = i + s - 1;
+					if(inp[i] == inp[k]) {
+						DP[i][k] = Math.min(DP[i][k], DP[i][j] + DP[j + 1][k] - 1);
```
Comment (SansPapyrus683) [inline solutions/usaco-1114.mdx:110]: comments would be nice

File: solutions/usaco-1114.mdx [hunk 5]
```diff
@@ -37,37 +37,88 @@ $$
...
 		}
 	}
 
-	for (int i = 0; i < n; i++) dp[i][i] = 1;
-	for (int s = 1; s <= n; s++) {
-		for (int i = 0; i <= n - s; i++) {
-			for (int j = i; j < i + s; j++) {
+	for(int i = 0; i < MAX_N; i++) dp[i][i] = 1;
+
+	for(int s = 1; s <= n; s++) {
```
Comment (SansPapyrus683) [inline solutions/usaco-1114.mdx:59]: same thing with the spaces and whatnot

File: solutions/usaco-1114.mdx [hunk 6]
```diff
@@ -37,37 +37,88 @@ $$
 ```cpp
 //CodeSnip{CPP Short Template}
 
-const int MAX_N = 300;
+const int MAX_N = 301;
```
Comment (SansPapyrus683) [inline solutions/usaco-1114.mdx:40]: wait why 301 300 should work perfectly fine

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
this is what i can find rn

File: solutions/usaco-1114.mdx (hunk 2 above)
Comment (SansPapyrus683) [reply solutions/usaco-1114.mdx]: nvm i fixed 1 myself

File: solutions/usaco-1114.mdx [hunk 7]
```diff
@@ -37,37 +37,85 @@ $$
...
 
+<JavaSection>
+	
+```java
+import java.io.*;
+import java.util.*;
+
+public class ModernArt3 {
+	public static final int MAXN = 301;
+	public static int[] inp = new int[MAXN];
```
Comment (SansPapyrus683) [inline solutions/usaco-1114.mdx:84]: again, could you rename this to `painting` or smth

**Comment by TruaShamu** [comment]
Alright, does this look more reasonable now, maybe?

**Review by SansPapyrus683 — APPROVED** [review]

**Comment by TruaShamu** [comment]
Wait, I broke something

**Comment by SansPapyrus683** [comment]
lmao f

**Comment by TruaShamu** [comment]
MAXN has to be 301 instead of 300 because the forloop runs for(i=0; i<=N; i++)
I'm an idiot :|

**Comment by TruaShamu** [comment]
So - can I change it back to 301...or...

**Comment by SansPapyrus683** [comment]
wait why does it have to be <= N

**Review by dongliu0426 — APPROVED** [review]

File: solutions/usaco-1114.mdx [hunk 8]
```diff
@@ -45,29 +45,80 @@ int main() {
...
 	for (int i = 0; i < n; i++) cin >> A[i];
+
 	for (int i = 0; i <= n; i++) {
 		for (int j = 0; j <= n; j++) {
 			dp[i][j] = MAX_N + 1;
 		}
 	}
 
-	for (int i = 0; i < n; i++) dp[i][i] = 1;
+	for (int i = 0; i < MAX_N; i++) dp[i][i] = 1;
```
Comment (dongliu0426) [inline solutions/usaco-1114.mdx:57]: actually, what's wrong with `i < n;`?

**Comment by TruaShamu** [comment]
Sorry, I was asleep - I'll go change both versions to < n

**Comment by TruaShamu** [comment]
Ok, done

**Comment by SansPapyrus683** [comment]
now does max n = 300 work?

**Comment by TruaShamu** [comment]
Yeah, it does!

For your viewing pleasure:

[Java AC](https://ibb.co/nzkc1W2)


[CPP AC](https://ibb.co/H491nhp)

**Comment by SansPapyrus683** [comment]
ok cool
now i can rest in pieces

**Pull request merged**